pkg/webui/oauth/views/update-password/index.js

```diff
@@ -0,0 +1,243 @@
+})
+
+const initialValues = {
+  password: '',
+  confirm: '',
+  revoke_all_access: true,
+}
+
+@connect(state => ({
+  fetching: state.user.fetching,
+  user: state.user.user,
+}), dispatch => ({
+  handleCancelUpdate: () => dispatch(push('/')),
+  handlePasswordChanged: () => dispatch(push('/login', {
+    info: m.passwordChanged,
+  })),
+  handleSessionRevoked: () => dispatch(push('/login', {
+    info: m.sessionExpired,
+  })),
+}))
+@bind
+export default class UpdatePassword extends React.PureComponent {
+
+  static propTypes = {
+    fetching: PropTypes.bool.isRequired,
+    user: PropTypes.object,
+    handleCancelUpdate: PropTypes.func.isRequired,
+    handleSessionRevoked: PropTypes.func.isRequired,
+    handlePasswordChanged: PropTypes.func.isRequired,
+  }
+
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      error: '',
+      info: '',
+      revoke_all_access: true,
+    }
+  }
+
+  handleRevokeAllAccess (evt) {
+    this.setState({ revoke_all_access: evt.target.checked })
+  }
+
+  async handleSubmit (values, { resetForm, setSubmitting }) {
+    const { user, handlePasswordChanged, handleSessionRevoked } = this.props
+    const userParams = queryString.parse(this.props.location.search)
+    const oldPassword = values.old_password ? values.old_password : userParams.current
+    const userId = Boolean(user) ? user.ids.user_id : userParams.user
+
+    try {
+      if (Boolean(user)) {
+        try {
+          await api.oauth.me()
+        } catch (error) {
+          handleSessionRevoked()
+          return
+        }
+      }
+      await api.users.updatePassword( userId, {
+        user_ids: { user_id: userId },
+        'new': values.password,
+        old: oldPassword,
+        revoke_all_access: values.revoke_all_access,
+      })
+      handlePasswordChanged()
+    } catch (error) {
+      this.setState({
+        error: error.response.data,
+        info: '',
+      })
+      setSubmitting(false)
+    }
+  }
+
+  render () {
+    const {
+      user,
+      fetching,
+      location,
+      handleCancelUpdate,
+    } = this.props
+
+    const {
+      error,
+      info,
+      revoke_all_access,
+    } = this.state
+
+    if (fetching) {
+      return (
+        <Spinner center>
+          <Message content={sharedMessages.loading} />
+        </Spinner>
+      )
+    }
+
+    const { user: userParam, current: currentParam } = queryString.parse(location.search)
+
+    if (!Boolean(user) && (!Boolean(userParam) || !Boolean(currentParam))) {
+      return (
+        <Redirect
+          to={{ pathname: '/' }}
+        />
+      )
+    }
+
+    let oldPasswordField
+    if (Boolean(user)) {
+      oldPasswordField = (
+        <Form.Field
+          component={Input}
+          required
+          title={m.oldPassword}
+          name="old_password"
+          type="password"
+          autoComplete="old-password"
+          autoFocus
+        />
+      )
+    }
+
+    var updatePasswordComponent = (
```

Comment:
dont use `var`s. 
once again: https://github.com/TheThingsNetwork/lorawan-stack/pull/1042#discussion_r306339487

```suggestion
    const updatePasswordComponent = (
```